src/test/java/io/jenkins/plugins/analysis/core/history/PreviousRunReferenceTest.java

```diff
@@ -0,0 +1,522 @@
+package io.jenkins.plugins.analysis.core.history;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import static io.jenkins.plugins.analysis.core.testutil.Assertions.assertThat;
+import io.jenkins.plugins.analysis.core.views.ResultAction;
+import static org.mockito.Mockito.*;
+
+import hudson.model.Result;
+import hudson.model.Run;
+
+/**
+ * PreviousRunReferenceTest to test PreviousRunReference.
+ */
+class PreviousRunReferenceTest extends ReferenceFinderTest {
+
+    @Override
+    ReferenceFinder getReferenceFinder(final Run baseline, final ResultSelector resultSelector) {
+        return new PreviousRunReference(baseline, resultSelector, true);
+    }
+
+    /**
+     * Test that there are no previous actions and stableRun is not build yet, with overallResultMusBeSuccess = false.
+     */
+    @Test
+    void notBuild_withAnyResult_shouldReturnNothing() {
+
+        Run<?, ?> baseline = mock(Run.class);
+        ResultSelector selector = mock(ResultSelector.class);
+        PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
+                false);
+
+        Optional<ResultAction> resultAction = previousRun.getReferenceAction();
+
+        assertThat(resultAction).isEmpty();
+    }
```

Comment:
Mr. Haffner mentioned, that we should rather use:
assertThat(previousRun.getReferenceAction()).isEmpty();